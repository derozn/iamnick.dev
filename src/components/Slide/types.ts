export type Props = {
  backgroundColor: string;
  delay?: number;
};
