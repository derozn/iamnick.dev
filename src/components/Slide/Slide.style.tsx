import styled, { keyframes } from 'styled-components';
import { Props } from './types';

const slide = keyframes`
  0% {
    transform-origin: right center;
    transform: scale3d(0, 1, 1);
  }

  50% {
    transform-origin: right center;
    transform: scale3d(1, 1, 1);
  }
  51% {
    transform-origin: left center;
  }

  100% {
    transform-origin: left center;
    transform: scale3d(0, 1, 1);
  }
`;

const show = keyframes`
  from {
    transform: translate3d(2%, 0, 0);
  }
  to {
    visibility: visible;
    transform: translate3d(0, 0, 0);
  }
`;

export const Container = styled.div<Props>`
  position: relative;

  &::after {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: ${({ backgroundColor }) => backgroundColor};
    transform: scale3d(0, 1, 1);
    animation: ${slide} 1s ${({ delay }) => (delay ? delay + 0.5 : 0.5)}s ease-in-out forwards;
  }

  > span {
    display: inherit;
    visibility: hidden;
    animation: ${show} 0.5s ${({ delay }) => (delay ? delay + 1 : 1)}s ease-out forwards;
  }
`;
