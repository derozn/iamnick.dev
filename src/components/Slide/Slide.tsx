import React, { FunctionComponent } from 'react';

import { Props } from './types';
import { Container } from './Slide.style';

const Slide: FunctionComponent<Props> = ({ children, ...rest }) => {
  return (
    <Container {...rest}>
      <span>{children}</span>
    </Container>
  );
};

export default Slide;
