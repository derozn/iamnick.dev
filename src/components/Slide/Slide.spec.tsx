import React from 'react';
import { render } from '@testing-library/react';

import Slide from './Slide';

describe('components/Slide', () => {
  it('renders children', () => {
    const { getByText } = render(<Slide backgroundColor="#fff">hello world!</Slide>);

    expect(getByText('hello world!')).toBeTruthy();
  });

  it('renders with background css style set to passed in backgroundColor', () => {
    const { container } = render(<Slide backgroundColor="#fff">hello world!</Slide>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with additional delay factor on slide animation', () => {
    const { container } = render(
      <Slide backgroundColor="#fff" delay={0.5}>
        hello world!
      </Slide>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
