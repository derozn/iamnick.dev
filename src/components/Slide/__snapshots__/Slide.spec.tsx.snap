// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Slide renders with additional delay factor on slide animation 1`] = `
.c0 {
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: #fff;
  -webkit-transform: scale3d(0,1,1);
  -ms-transform: scale3d(0,1,1);
  transform: scale3d(0,1,1);
  -webkit-animation: hDiCPJ 1s 1s ease-in-out forwards;
  animation: hDiCPJ 1s 1s ease-in-out forwards;
}

.c0 > span {
  display: inherit;
  visibility: hidden;
  -webkit-animation: cwLUPY 0.5s 1.5s ease-out forwards;
  animation: cwLUPY 0.5s 1.5s ease-out forwards;
}

<div
  class="c0"
>
  <span>
    hello world!
  </span>
</div>
`;

exports[`components/Slide renders with background css style set to passed in backgroundColor 1`] = `
.c0 {
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: #fff;
  -webkit-transform: scale3d(0,1,1);
  -ms-transform: scale3d(0,1,1);
  transform: scale3d(0,1,1);
  -webkit-animation: hDiCPJ 1s 0.5s ease-in-out forwards;
  animation: hDiCPJ 1s 0.5s ease-in-out forwards;
}

.c0 > span {
  display: inherit;
  visibility: hidden;
  -webkit-animation: cwLUPY 0.5s 1s ease-out forwards;
  animation: cwLUPY 0.5s 1s ease-out forwards;
}

<div
  class="c0"
>
  <span>
    hello world!
  </span>
</div>
`;
